File: frontend/src/services/discoverService.ts
import * as DiscoverBinding from '../../bindings/changeme/backend/services/discoverservice.js'
import type { DiscoverFeature } from '../types'
import { mockFeatures, shouldUseMock } from './mockFallback'

export const discoverService = {
    async getFeatures(): Promise<DiscoverFeature[]> {
        if (shouldUseMock()) return mockFeatures
        const result = await DiscoverBinding.GetFeatures()
        return (result ?? []) as unknown as DiscoverFeature[]
    },
}

File: backend/services/discover_service.go
package services

import (
	"changeme/backend/domain"
	"changeme/backend/repository"
)

// DiscoverService 提供发现 Tab 的功能列表
type DiscoverService struct {
	repo repository.DiscoverRepository
}

func NewDiscoverService(r repository.DiscoverRepository) *DiscoverService {
	return &DiscoverService{repo: r}
}

// GetFeatures 返回发现 Tab 的功能项
func (s *DiscoverService) GetFeatures() ([]domain.DiscoverFeature, error) {
	return s.repo.ListFeatures()
}
